Model/PaymentLibrary.php

```diff
@@ -0,0 +1,487 @@
+        ScopeConfigInterface $scopeConfig,
+        Logger $logger,
+        ConfigRepository $configRepository,
+        GingerClient $gingerClient,
+        ProcessTransactionRequest $processTransactionRequest,
+        ProcessTransactionUpdate $processTransactionUpdate,
+        OrderLines $orderLines,
+        OrderDataCollector $orderDataCollector,
+        CustomerData $customerData,
+        Session $checkoutSession,
+        Order $order,
+        GetOrderByTransaction $getOrderByTransaction,
+        UrlProvider $urlProvider,
+        ManagerInterface $messageManager,
+        AbstractResource $resource = null,
+        AbstractDb $resourceCollection = null,
+        array $data = []
+    ) {
+        parent::__construct(
+            $context,
+            $registry,
+            $extensionFactory,
+            $customAttributeFactory,
+            $paymentData,
+            $scopeConfig,
+            $logger,
+            $resource,
+            $resourceCollection,
+            $data
+        );
+        $this->configRepository = $configRepository;
+        $this->gingerClient = $gingerClient;
+        $this->processTransactionRequest = $processTransactionRequest;
+        $this->processTransactionUpdate = $processTransactionUpdate;
+        $this->customerData = $customerData;
+        $this->orderLines = $orderLines;
+        $this->orderDataCollector = $orderDataCollector;
+        $this->checkoutSession = $checkoutSession;
+        $this->order = $order;
+        $this->getOrderByTransaction = $getOrderByTransaction;
+        $this->urlProvider = $urlProvider;
+        $this->messageManager = $messageManager;
+    }
+
+    /**
+     * Extra checks for method availability
+     *
+     * @param CartInterface|null $quote
+     *
+     * @return bool
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function isAvailable(CartInterface $quote = null)
+    {
+        $client = $this->loadGingerClient();
+
+        if(!$this->checkoutSession->getMultiCurrency())
+        {
+            try
+            {
+                $this->checkoutSession->setMultiCurrency($client->getCurrencyList());
+            }
+            catch (Exception $exception)
+            {
+                if (strstr($exception->getMessage(),"Forbidden(403)"))
+                {
+                    $this->checkoutSession->setMultiCurrency(null);
+                }
+            }
+        }
+
+        if($this->checkoutSession->getMultiCurrency())
+        {
+            if (array_key_exists($this->platform_code, $this->checkoutSession->getMultiCurrency()['payment_methods']))
+            {
+                $currencyForCurrentPayment = $this->checkoutSession->getMultiCurrency()['payment_methods'][$this->platform_code]['currencies'];
+            }
+            else
+            {
+                return false;
+            }
+        }
+        else
+        {
+            $currencyForCurrentPayment = ['EUR'];
+        }
```

Comment:
unnecessary 'else' statement that can be replaced by 'early return', for example:

```
if($this->checkoutSession->getMultiCurrency()) {
    if (!array_key_exists($this->platform_code, $this->checkoutSession->getMultiCurrency()['payment_methods'])) return false;
    $currencyForCurrentPayment = $this->checkoutSession->getMultiCurrency()['payment_methods'][$this->platform_code]['currencies'];
  } else {
      $currencyForCurrentPayment = ['EUR'];
  }
```

Reply:
But it's still looks like it can be simplified another one time, probably you find way to avoid 'else' statement